### ISSUE TITLE
Code splitting with default exports produces broken CJS output

### ISSUE THREAD

**Issue by nullobject** [rollup/rollup#2721]
- Rollup Version: 1.3.0
- Operating System (or Browser): Mac OS Mojave
- Node Version: 10.14.0

### How Do We Reproduce?

Code splitting any modules with default exports will produce broken CJS output, even for very simple modules. I noticed the issue when reading [this blog post](https://levelup.gitconnected.com/code-splitting-for-libraries-bundling-for-npm-with-rollup-1-0-2522c7437697) by @lukastaegert.

Say we have two files:

```js
// foo.js
export default 'hello'
```

```js
// main.js
export { default as foo } from './foo'
```

When these files are bundled with code splitting we get:

```js
// foo.js
module.exports = 'hello'
```

```js
// main.js
var foo = require('./foo.js')
exports.foo = foo.default
```

But `foo.default` is undefined!

The blog post I mentioned earlier has a [REPL](https://rollupjs.org/repl?gist=e9c6c04b8f96adc562a70c096c3e7705&entry=lower.js,upper.js,upperFirst.js) which illustrates this issue. Just look at the CJS output and you'll see the export statements in `main.js` are broken.

### Expected Behavior

```js
// main.js
var foo = require('./foo.js');
exports.foo = foo;
```

### Actual Behavior

```js
// main.js
var foo = require('./foo.js');
exports.foo = foo.default;
```

**Comment by lukastaegert** [comment]
Thanks for spotting this. Just some notes to myself or whoever fixes this (I hope to find time soon but cannot make promises about the next days):
- we could just add the normal interop (not nice)
- much better: we make sure the chunk has the correct exports mode and dynamically determine from there how to retrieve the default

**Comment by lukastaegert** [comment]
Fix at #2727

**Comment by nullobject** [comment]
Legend, thank you ❤️
